src/Compilers/CSharp/Portable/Syntax/SyntaxNodeRemover.cs

```diff
@@ -465,10 +465,32 @@ private void AddDirectives(SyntaxNode node, TextSpan span)
 
                         if (_directivesToKeep.Contains(directive))
                         {
-                            AddResidualTrivia(SyntaxFactory.TriviaList(directive.ParentTrivia), requiresNewLine: true);
+                            var parentTrivia = directive.ParentTrivia;
+                            var (triviaList, triviaListIndex) = getTriviaListAndIndex(parentTrivia);
```

Comment:
I think being explicit that this is the index of the directive will help readability here.
```suggestion
                            var (triviaList, directiveTriviaListIndex) = getTriviaListAndDirectiveIndex(parentTrivia);
```

Reply:
done.